source/extensions/filters/common/set_filter_state/filter_config.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include "envoy/extensions/filters/common/set_filter_state/v3/value.pb.h"
+#include "envoy/formatter/substitution_formatter.h"
+#include "envoy/stream_info/filter_state.h"
+
+#include "source/common/common/logger.h"
+#include "source/common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace SetFilterState {
+
+using LifeSpan = StreamInfo::FilterState::LifeSpan;
+using StateType = StreamInfo::FilterState::StateType;
+using StreamSharing = StreamInfo::StreamSharingMayImpactPooling;
+
+struct Value {
+  std::string key_;
+  StreamInfo::FilterState::ObjectFactory* factory_;
+  StateType state_type_{StateType::ReadOnly};
+  StreamSharing stream_sharing_{StreamSharing::None};
+  bool skip_if_empty_;
+  Formatter::FormatterConstSharedPtr value_;
+};
+
+class Config : public Logger::Loggable<Logger::Id::config> {
+public:
+  Config(
+      const Protobuf::RepeatedPtrField<
+          envoy::extensions::filters::common::set_filter_state::v3::FilterStateValue>& proto_values,
```

Comment:
nit: a alias name of type `envoy::extensions::filters::common::set_filter_state::v3::FilterStateValue` like `FilterStateValueProto` may be helpful to the readability.

Reply:
Done